tools/docvalidator/main.go

```diff
@@ -0,0 +1,165 @@
+// Package main provides a utility to process markdown documentation files,
+// extract code examples, execute them, and verify their outputs.
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log/slog"
+	"os"
+	"strings"
+	"time"
+)
+
+// Example represents an individual code example extracted from the markdown files.
+// It contains the function name, the file it was extracted from, the code snippet, and the expected output.
+type Example struct {
+	FuncName string
+	File     string
+	Code     string
+	Expected string
+	Skipped  bool
+}
+
+var logLevel string
+
+// init initializes the logging level from command-line flags.
+// It sets the default logging level to "info", but can be overridden by
+// specifying a different level (debug, info, warn, error) using the -log flag.
+func init() {
+	flag.StringVar(&logLevel, "log", "info", "Set the logging level (debug, info, warn, error)")
```

Comment:
Please define a constant []string containing all the constants I suggest you to created, and here use `strings.Join(", ", levels)`

https://github.com/go-sprout/sprout/pull/86#discussion_r1861219669